lexos/models/content_analysis_model.py

```diff
@@ -0,0 +1,481 @@
+
+        :return: formula_errors: str with error message or None if there is no
+                 errors.
+                 result_table: str containing result html table or None if
+                 there are any errors in the formula
+        """
+        self.count()
+        if self.is_secure():
+            formula_errors = self.save_formula()
+            self.generate_scores()
+            self.generate_averages()
+            result_table = self.to_html()
+        else:
+            formula_errors = "Formula error: Invalid input"
+            result_table = ""
+        return result_table, formula_errors
+
+    @property
+    def dictionaries(self) -> list:
+        """:return: dictionaries"""
+        return self._dictionaries
+
+    @dictionaries.setter
+    def dictionaries(self, dictionaries: list):
+        """Set dictionaries."""
+        self._dictionaries = dictionaries
+
+    @property
+    def corpus(self) -> list:
+        """:return: corpus"""
+        return self._corpus
+
+    @property
+    def counters(self) -> list:
+        """:return: counters"""
+        return self._counters
+
+    @property
+    def scores(self) -> list:
+        """Sets scores."""
+        return self._scores
+
+    @property
+    def averages(self) -> list:
+        """:return: averages"""
+        return self._averages
+
+    @property
+    def content_analysis_option(self) -> ContentAnalysisOption:
+        """Gets front-end options from ContentAnalysisReceiver.
+
+        :return: front-end or test options
+        """
+        if self._test_options is not None:
+            if self._test_options.formula is not None:
+                self.save_formula()
+            return self._test_options
+        return ContentAnalysisReceiver().options_from_front_end()
+
+    @property
+    def test_option(self):
+        """:return: test options"""
+        return self._test_options
+
+    @test_option.setter
+    def test_option(self, options):
+        """Sets test options."""
+        self._test_options = options
+
+
+def count_phrases(dictionary: list, file: object):
+    """Counts each phrase in the dictionary in the given file.
+
+    If a has with more than 1 word is found in a file, it is deleted
+    from the file to prevent double count.
+    For example:
+    dictionary[1].content = "not very good"
+    dictionary[2].content = "very good"
+    file.content = "not very good"
+    count: "not very good" = 1
+           "very good" = 0
+    :param dictionary: list of Phrase objects sorted by number of word in
+    descending order
+    :param file: a File object
+    :return: list of Phrase objects with their counts
+    """
+    for phrase in dictionary:
+        phrase.count = count_phrase_in_text(phrase=phrase.content,
+                                            text=file.content)
+        if ' ' in phrase.content:
+            file.content = file.content.replace(phrase.content, ' ')
```

Comment:
I don't think this logic is correct... 

If I have phrase "not very good", and phrase "very   good", and I have test "very not very good good", then both phrases will be matched?

Reply:
The "very good" has 3 spaces between them. Somehow GitHub removes the 2 spaces.

Reply:
Again, why would anyone put "very   good" instead of "very good"? Also, the only way I can think of fixing this is replacing the found phrase with some kind of tag, but there is always the possibility of that tag been in the dictionary anyway.

Reply:
same as https://github.com/WheatonCS/Lexos/pull/660#discussion-diff-155119410R10